code/datums/outfits/outfit.dm

```diff
@@ -136,36 +170,37 @@
 			if(equip_holster.loc != equip_uniform && !QDELETED(equip_holster))
 				qdel(equip_holster)
 
-	if(suit)
-		wearer.equip_to_slot_or_del(new suit(wearer),slot_wear_suit_str)
-	if(back)
-		wearer.equip_to_slot_or_del(new back(wearer),slot_back_str)
-	if(belt)
-		wearer.equip_to_slot_or_del(new belt(wearer),slot_belt_str)
-	if(gloves)
-		wearer.equip_to_slot_or_del(new gloves(wearer),slot_gloves_str)
-	if(shoes)
-		wearer.equip_to_slot_or_del(new shoes(wearer),slot_shoes_str)
-	if(mask)
-		wearer.equip_to_slot_or_del(new mask(wearer),slot_wear_mask_str)
-	if(head)
-		wearer.equip_to_slot_or_del(new head(wearer),slot_head_str)
 	if(l_ear)
-		var/l_ear_path = (OUTFIT_ADJUSTMENT_PLAIN_HEADSET & equip_adjustments) && ispath(l_ear, /obj/item/radio/headset) ? /obj/item/radio/headset : l_ear
-		wearer.equip_to_slot_or_del(new l_ear_path(wearer),slot_l_ear_str)
+		for(var/l_ear_type in resolve_equip_to_list(l_ear))
+			var/l_ear_path = (OUTFIT_ADJUSTMENT_PLAIN_HEADSET & equip_adjustments) && ispath(l_ear_type, /obj/item/radio/headset) ? /obj/item/radio/headset : l_ear_type
+			try_equip(wearer, l_ear_path, slot_l_ear_str)
 	if(r_ear)
-		var/r_ear_path = (OUTFIT_ADJUSTMENT_PLAIN_HEADSET & equip_adjustments) && ispath(r_ear, /obj/item/radio/headset) ? /obj/item/radio/headset : r_ear
-		wearer.equip_to_slot_or_del(new r_ear_path(wearer),slot_r_ear_str)
-	if(glasses)
-		wearer.equip_to_slot_or_del(new glasses(wearer),slot_glasses_str)
-	if(id)
-		wearer.equip_to_slot_or_del(new id(wearer),slot_wear_id_str)
-	if(l_pocket)
-		wearer.equip_to_slot_or_del(new l_pocket(wearer),slot_l_store_str)
-	if(r_pocket)
-		wearer.equip_to_slot_or_del(new r_pocket(wearer),slot_r_store_str)
-	if(suit_store)
-		wearer.equip_to_slot_or_del(new suit_store(wearer),slot_s_store_str)
+		for(var/r_ear_type in resolve_equip_to_list(r_ear))
+			var/r_ear_path = (OUTFIT_ADJUSTMENT_PLAIN_HEADSET & equip_adjustments) && ispath(r_ear_type, /obj/item/radio/headset) ? /obj/item/radio/headset : r_ear_type
+			try_equip(wearer, r_ear_path, slot_r_ear_str)
+
+	// Ears, backpack and uniforms are handled individually.
+	var/list/slot_to_gear  = list(
+		slot_wear_suit_str = resolve_equip_to_list(suit),
```

Comment:
this is a define, so it might actually work, but i'm a little worried about using bare list keys like this. actually no i know for a fact it'll work until someone makes a typo or include order error when adding a new one, and then it'll insert the nonexistent variable as a string. so... dunno. fine as is i guess but wrapping it in parens is technically safer